### ISSUE TITLE
About multi-labels

### ISSUE THREAD

**Issue by sousuodaren** [thunlp/OpenNRE#12]
In the paper: [Lin et al., 2016] Yankai Lin, Shiqi Shen, Zhiyuan Liu, Huanbo Luan, and Maosong Sun. Neural Relation Extraction with Selective Attention over Instances. In Proceedings of ACL,
did the author take account into multi labels problem?
Can you tell me how did you fix bugs of multi-label ? I have not found that you have modified the loss function and prediction.

**Comment by Tsingularity** [comment]
1.In the ACL paper, did the author take account into multi labels problem?

Yes, the author set different attention query vectors for different relations.

2.How did I fix bugs of multi-label?

In fact, the previous bug of multi-label in my code of former version refers to the initial.py. When reading sentences and labels for the dataset, I made a mistake that I ignored the same entity pairs could have more than one relations.
In the code, at present, I use the single query vector method for the sentence level attention. During the experiments, I also tried the multi-label query vectors method as described in the paper, but it turned out that it didn't achieve as good results as the single query vector method, which is also quite confusing for me.
At present, I am also trying all kinds of techniques to improve the performance of multi-label query vector method on the dataset.

**Comment by ShomyLiu** [comment]
@Tsingularity  
Hi, 

Have you reproduced the performance of multi-label query vector used in Lin 2016? I have tried an implement in pytorch of Lin 2016 with multi-label attention but can not achieve the result quite, which confused me a lot. 

Thanks.
